### ISSUE TITLE
Bug: SingleFile method fail (run forever) with new Chromium versions (121.0.6167.139-1~deb12u1) on Debian 12

### ISSUE THREAD

**Issue by codeurimpulsif** [ArchiveBox/ArchiveBox#1502]
#### Describe the bug

SingleFile method fail with new Chromium versions. The issue is here since less than 2 month I think.

I'm on a Debian 12.6 (aarch64) with Chromium version `128.0.6613.84-1~deb12u1`.

If I downgrade Chromium to version `121.0.6167.139-1~deb12u1` everything works as expected.

So maybe something changed in Chromium causing this issue.

#### Steps to reproduce

1. Ran ArchiveBox with a config allowing SingleFile method with Chromium version `128.0.6613.84-1~deb12u1`
2. Add a webpage to snapshot (via CLI or webUI)
3. Watch Chromium running forever without exiting and the webUI go 404
4. Kill Chromium process
5. Downgrade `chromium` (and `chromium-common` of course) package to version `121.0.6167.139-1~deb12u1`
6. Add the same webpage to snapshot
7. Watch Chromium running for less than a minute and the page is successfully snapshoted

#### Screenshots or log output

The `errors.log` don't show anything.

#### ArchiveBox version

```logs
$ archivebox version
0.7.2
ArchiveBox v0.7.2 BUILD_TIME=2024-02-03 00:33:38 1706916818
IN_DOCKER=False IN_QEMU=False ARCH=aarch64 OS=Linux PLATFORM=Linux-6.1.21-v8+-aarch64-with-glibc2.36 PYTHON=Cpython
FS_ATOMIC=True FS_REMOTE=False FS_USER=995:991 FS_PERMS=644
DEBUG=False IS_TTY=True TZ=UTC SEARCH_BACKEND=ripgrep LDAP=False

[i] Dependency versions:
 √  PYTHON_BINARY         v3.11.2         valid     /usr/bin/python3.11
 √  SQLITE_BINARY         v2.6.0          valid     /usr/lib/python3.11/sqlite3/dbapi2.py
 √  DJANGO_BINARY         v3.1.14         valid     /data/1/webarchive/venv/lib/python3.11/site-packages/django/__init__.py
 √  ARCHIVEBOX_BINARY     v0.7.2          valid     /data/1/webarchive/venv/bin/archivebox

 √  CURL_BINARY           v7.88.1         valid     /usr/bin/curl
 -  WGET_BINARY           -               disabled  /usr/bin/wget
 √  NODE_BINARY           v18.19.0        valid     /usr/bin/node
 √  SINGLEFILE_BINARY     v1.0.24         valid     ./node_modules/single-file/cli/single-file
 -  READABILITY_BINARY    -               disabled  ./node_modules/readability-extractor/readability-extractor
 -  MERCURY_BINARY        -               disabled  postlight-parser
 -  GIT_BINARY            -               disabled  /usr/bin/git
 -  YOUTUBEDL_BINARY      -               disabled  /data/1/webarchive/venv/bin/yt-dlp
 √  CHROME_BINARY         v121.0.6167.139  valid     /usr/bin/chromium
 √  RIPGREP_BINARY        v13.0.0         valid     /usr/bin/rg

[i] Source-code locations:
 √  PACKAGE_DIR           23 files        valid     /data/1/webarchive/venv/lib/python3.11/site-packages/archivebox
 √  TEMPLATES_DIR         3 files         valid     /data/1/webarchive/venv/lib/python3.11/site-packages/archivebox/templates
 -  CUSTOM_TEMPLATES_DIR  -               disabled  None

[i] Secrets locations:
 -  CHROME_USER_DATA_DIR  -               disabled  None
 -  COOKIES_FILE          -               disabled  None

[i] Data locations:
 √  OUTPUT_DIR            6 files         valid     /data/1/webarchive/archivebox
 √  SOURCES_DIR           56 files        valid     ./sources
 √  LOGS_DIR              1 files         valid     ./logs
 √  ARCHIVE_DIR           37 files        valid     ./archive
 √  CONFIG_FILE           442.0 Bytes     valid     ./ArchiveBox.conf
 √  SQL_INDEX             476.0 KB        valid     ./index.sqlite3
```

**Comment by pirate** [comment]
Thanks for reporting.

We are using an older version of singlefile (1.5.x) that is no longer supported, it's likely fixed in their more recent 2.x.x releases.

The reason we haven't already updated to 2.x.x. is because single-file-cli switched from using node to deno, so it requires a lot of work (currently ongoing in the v0.8.x `dev` branch) to adadpt ArchiveBox's packaging system.

If the old singlefile is now failing that means I now have added urgency to finish this work!

**Comment by pirate** [comment]
Actually I noticed in your log that you're using an even older version of singe-file instead of the latest known good version `1.1.54`. You should try updating to a slightly newer version first:

```bash
cd /data/1/webarchive/archivebox

npm install single-file-cli@1.1.54
```

**Comment by codeurimpulsif** [comment]
Oh you're right, I forgot to update node dependencies!

Running `archivebox setup` again fix the issue with Chromium `128.0.6613.84-1~deb12u1`.

Thanks for your help.

Feel free to close this issue if it's not relevant anymore.
